scripts/main.js

```diff
@@ -52,20 +52,32 @@ $appList.on('click', 'li', function (event) {
 $goBackButton.on('click', function () {
   setRoute('')
 })
-
-
 $editButton.on('click', function(event){
-$body.attr('data-state', 'update-app')
 var id = $('#name-app').data('id')
 setRoute(id + '/edit')
-handleEditRout()
+})
+
 $updateAppForm.on('submit', function (event) {
   event.preventDefault()
-  setRoute(id)
-  })
-  $cancelButton.on('click',function(event){
+
+  var id = $('#edit-app-container').data('id')
```

Comment:
instead of embedding the `$updateAppForm` submit handler into the `$editButton` click handler, we should assign this event only once and read the current id from the edit app container.

Before that, we added another `submit` handler each time we clicked the edit button, so the code in the submit handler would run multiple times